app/assets/javascripts/map.js

```diff
@@ -1,152 +1,165 @@
 //= require polyline
 
-      var directionsDisplay;
-      var directionsService = new google.maps.DirectionsService();
-      var map;
+var directionsService = new google.maps.DirectionsService();
 
-      function initialize() {
-        directionsDisplay = new google.maps.DirectionsRenderer();
+function initialize() {
+  directionsDisplay = new google.maps.DirectionsRenderer();
 
-        var mapOptions = {
-          zoom: 4,
-          mapTypeId: google.maps.MapTypeId.ROADMAP
-        };
+  var mapOptions = {
+    zoom: 4,
+    mapTypeId: google.maps.MapTypeId.ROADMAP
+  };
 
-        map = new google.maps.Map(document.getElementById('map_canvas'), mapOptions);
-        directionsDisplay.setMap(map);
+  map = new google.maps.Map(document.getElementById('map_canvas'), mapOptions);
+  directionsDisplay.setMap(map);
+  route = new Route();
+  route.calculateRoute();
 
-        calcRoute();
-      }
+}
 
-      function calcRoute() {
+//-------Route----------
 
-        var start = new google.maps.LatLng(41.953819, -87.654750); // Chicago
-        var end = new google.maps.LatLng(38.637548, -90.205010); // St. Louis
-        var waypts = [];
-        var checkboxArray = document.getElementById('waypoints');
+var Route = function(){
+  this.start = new google.maps.LatLng(41.953819, -87.654750), // Chicago for now
+  this.end = new google.maps.LatLng(38.637548, -90.205010), // St. Louis for now
+  this.waypts = this.getWaypoints();
+};
 
-        for (var i = 0; i < checkboxArray.length; i++) {
-          if (checkboxArray.options[i].selected == true) {
-            waypts.push({
-              location:checkboxArray[i].value,
-              stopover:true});
-          }
-        }
+Route.prototype.getWaypoints = function(){
+  var waypts = [];
+  var checkboxArray = document.getElementById('waypoints');
```

Comment:
From an OO perspective, your `Route` class is now interested in the status of the DOM, which is certainly a view concern.  This DOM interaction should be bubbled out of `Route` and, given the present code, be handled after `getWaypoints` is called (i.e. in your `initialziation` function).